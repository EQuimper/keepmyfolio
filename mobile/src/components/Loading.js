// @flow

import React from 'react';
import { ActivityIndicator, View, StyleSheet } from 'react-native';

/**
 * UTILS
 */
import { colors } from '../utils/constants';

const styles = StyleSheet.create({
  root: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'center',
  },
});

type Props = {
  color?: string,
  size?: 'large' | 'small',
};

function Loading({ color = colors.primary, size = 'large' }: Props = {}) {
  return (
    <View style={styles.root}>
      <ActivityIndicator color={color} size={size} />
    </View>
  );
}

export default Loading;
