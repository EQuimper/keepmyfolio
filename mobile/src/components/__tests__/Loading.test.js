// @noflow

import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Loading from '../Loading';

describe('Loading Components', () => {
  it('should renders correctly', () => {
    const component = renderer.create(<Loading />).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('should renders black if pass color black', () => {
    const component = renderer.create(<Loading color="#000" />).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('should be small if pass size small', () => {
    const component = renderer.create(<Loading size="small" />).toJSON();

    expect(component).toMatchSnapshot();
  });
});
