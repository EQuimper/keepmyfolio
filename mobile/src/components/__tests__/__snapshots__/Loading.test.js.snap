// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Components should be small if pass size small 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#00C0FF"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;

exports[`Loading Components should renders black if pass color black 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#000"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`Loading Components should renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#00C0FF"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
